doc/decisions/map_preloading.md

```diff
@@ -0,0 +1,51 @@
+# Map Preloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects. However, not all layers are visible simultaneously, nor are they required during the initial loading of the application. Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to preload data on the map to optimize its performance.
+
+## Constraints
+
+1.
+
+## Assumptions
+
+1.
+
+## Solutions
+
+### Full fetch
+
+Load all the data from the backend on the initial load. This will result in a slower initial load, but the map will be ready to use immediately.
+
+### Pagination
+
+Since the canvas data is stored in a nested manner, we cannot simply fetch the data in chunks as we would do it with a flat array structure.
```

Comment:
I don't understand the sentence. Is this about the preload of just the currently visible screen? I don't see a big problem that we only fetch data for parts of the map, PostGIS should be very capable of doing this (if we design tables properly, if we have polygons over the whole map, obviously the whole polygon needs to be loaded). (@NurramoX What do you think?)

Reply:
This solution relates to the preloading topic. I rephrased the sentence, so it should be clear now that it is not about the visible screen part but about canvas elements' database structure, i.e. fetch only the first 20 pages of the table in an online shop list. The case is not applicable in our case due to the nested structure of the canvas.